From 6ccc4326c253d9cd492ec6052b555994a2e1ee6e Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Sat, 03 Jun 2023 08:32:17 +0300
Subject: [PATCH] [SW-145263] h6: sec: fix pcie link issue in secure setup

with recent changes of fake security alignmnent with real security
pcie link up time is increased as the bis compliant mode is checked
before pcie link up. So cant have any prints before link up
Updated code to remove the print and also corrected error condition

Change-Id: I9d080d5a9322f1618e63e88d54ca703e8b830257
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 5099175..2826594 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -1586,10 +1586,8 @@
 
 	if (preboot_security_is_real_efuse_sec_en()) {
 		fuse = preboot_get_hl_sec_real_efuse();
-		if (fuse) {
-			preboot_puts("failed to read efuse, forcing bis compliant mode.\n");
+		if (!fuse)
 			return true;
-		}
 
 		return fuse->bis_compliant;
 	}
